### PR TITLE
Eigsolve AD

### PR DESCRIPTION
**Pull request by lkdvos**
Almost working version of the eigsolve rrules.
Seems to be failing only for real matrices, where comparison with rrule for eigen does not give equal results.
Specifically, it seems that the current rrule has the exact same real part, but different imaginary part.

### CODE REVIEW

**Comment by Jutho** [comment]
Cool; I'll try to review as soon as possible.

**Comment by maartenvd** [comment]
> Seems to be failing only for real matrices, where comparison with rrule for eigen does not give equal results.

is the problem the sorting, or is there something mathematically wrong?

**Comment by lkdvos** [comment]
> > Seems to be failing only for real matrices, where comparison with rrule for eigen does not give equal results.
> 
> is the problem the sorting, or is there something mathematically wrong?

I am just guessing, but I don't see what else could be different for real matrices

File: src/eigsolve/eigsolve.jl [hunk 1]
```diff
@@ -292,3 +293,15 @@ function eigsort(which::Symbol)
...
+    if which == :LI
+        return :SI
+    elseif which == :SI
+        return :LI
+    else
+        return which
+    end
+end
+
+conjwhich(s::EigSorter) = EigSorter(conj(s.by), s.rev)
```
Comment (Jutho) [inline src/eigsolve/eigsolve.jl:307]: `s.by` can be a function, so I don't see how that would work? Do you mean something like `x->s.by(conj(x))` ?

Comment (lkdvos) [reply src/eigsolve/eigsolve.jl]: ah, my mistake, I only  tested with ```:LM``` so this escaped from my attention, should be fixed now

**Comment by Jutho** [comment]
Closed in favor of #85

**Pull request closed without merging**